### PR TITLE
docs: API users

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/3522

Blocker: https://github.com/coder/coder/pull/5546 (this PR is built on top of that branch)

Changes:
* users

### CODE REVIEW

File: cli/userstatus_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func TestUserStatus(t *testing.T) {
 	})
 
 	t.Run("StatusOther", func(t *testing.T) {
-		require.Equal(t, otherUser.Status, codersdk.UserStatusActive, "start as active")
+		require.Equal(t, codersdk.UserStatusActive, otherUser.Status, "start as active")
```
Comment (mafredri) [inline cli/userstatus_test.go:36]: Nice catch, this is a downside of `require`. 😅

File: docs/api/authorization.md [hunk 2]
```diff
@@ -67,3 +67,50 @@ curl -X POST http://coder-server:8080/api/v2/authcheck \
...
+  -H 'Coder-Session-Token: API_KEY'
+```
+
+`POST /users/login`
+
+> Body parameter
+
+```json
+{
+  "email": "string",
```
Comment (mafredri) [inline docs/api/authorization.md:89]: Missing email format from `codersdk.LoginWithPasswordRequest`?

File: coderd/apikey.go [hunk 3]
```diff
@@ -25,6 +25,16 @@ import (
...
+//
+// @Summary Create token API key
+// @ID create-token-api-key
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Users
+// @Param user path string true "User ID, name, or me"
+// @Param request body codersdk.CreateTokenRequest true "Create token request"
+// @Success 201 {object} codersdk.GenerateAPIKeyResponse
+// @Router /users/{user}/keys/tokens [post]
```
Comment (mafredri) [inline coderd/apikey.go:37]: I recently learned about `swag fmt`.
```
go run github.com/swaggo/swag/cmd/swag@latest fmt --dir ./coderd
```
Should we enable it for our repo? It's pretty fast.

One gotcha though: Every function needs a proper comment body or go fmt will undo some of the changes.

Example:

`swag fmt`:
```go
//	@Summary	Get audit logs
//	@ID			get-audit-logs
//	@Security	CoderSessionToken
//	@Produce	json
//	@Tags		Audit
//	@Param		q			query		string	true	"Search query"
//	@Param		after_id	query		string	false	"After ID"	format(uuid)
//	@Param		limit		query		int		false	"Page limit"
//	@Param		offset		query		int		false	"Page offset"
//	@Success	200			{object}	codersdk.AuditLogResponse
//	@Router		/audit [get]
```
`go fmt`:
```go
// @Summary	Get audit logs
// @ID			get-audit-logs
// @Security	CoderSessionToken
// @Produce	json
// @Tags		Audit
// @Param		q			query		string	true	"Search query"
// @Param		after_id	query		string	false	"After ID"	format(uuid)
// @Param		limit		query		int		false	"Page limit"
// @Param		offset		query		int		false	"Page offset"
// @Success	200			{object}	codersdk.AuditLogResponse
// @Router		/audit [get]
```

(Notice tab prefix -> space.)

However, with a comment it's fine.
```go
// Get audit logs.
//
//	@Summary	Get audit logs
//	@ID			get-audit-logs
//	@Security	CoderSessionToken
//	@Produce	json
//	@Tags		Audit
//	@Param		q			query		string	true	"Search query"
//	@Param		after_id	query		string	false	"After ID"	format(uuid)
//	@Param		limit		query		int		false	"Page limit"
//	@Param		offset		query		int		false	"Page offset"
//	@Success	200			{object}	codersdk.AuditLogResponse
//	@Router		/audit [get]
```

I dislike the full-on tab indentation though (will be messy for different settings of editor indentation display). We could submit a PR to `swaggo/swag` to use space indentation for everything _except_ the prefix. Related PR that changed the behavior: https://github.com/swaggo/swag/pull/1386 from all space to all tab (vs just fixing the prefix).

**Review by mafredri — APPROVED** [review]
Super, nice job!

File: coderd/apikey.go (hunk 3 above)
Comment (mtojek) [reply coderd/apikey.go:37]: I can enable it in a separate PR to cover all change, but I'm concerned if it doesn't conflict with the project formatter.

File: docs/api/authorization.md (hunk 2 above)
Comment (mtojek) [reply docs/api/authorization.md]: Fixed

**Pull request merged**